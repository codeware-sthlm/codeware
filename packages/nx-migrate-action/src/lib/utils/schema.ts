import { z } from 'zod';

export const EmailValidation = {
  standardEmail: z.string().email(),
  githubBotEmail: z
    .string()
    .regex(
      /^\d+\+[\w-]+\[bot\]@users\.noreply\.github\.com$/,
      'Invalid GitHub bot email'
    ),
  anyBotEmail: z.string().refine(
    (email) => {
      const githubBotPattern =
        /^\d+\+[\w-]+\[bot\]@users\.noreply\.github\.com$/;
      const regularEmailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
      return githubBotPattern.test(email) || regularEmailPattern.test(email);
    },
    {
      message: 'Invalid email address'
    }
  )
};

export const ActionInputsSchema = z.object({
  author: z.string(),
  autoMerge: z.boolean(),
  checkToken: z.boolean(),
  committer: z.string(),
  dryRun: z.boolean(),
  mainBranch: z.string(),
  packagePatterns: z.array(z.string()),
  prAssignees: z.string(),
  skipTests: z.boolean(),
  skipE2E: z.boolean(),
  token: z.string().min(1, 'token is required')
});

export const MigrateConfigSchema = z.object({
  author: z
    .object({
      name: z.string(),
      email: EmailValidation.anyBotEmail
    })
    .optional(),
  autoMerge: z.boolean(),
  committer: z.object({
    name: z.string().min(1, 'committer name is required'),
    email: EmailValidation.anyBotEmail
  }),
  dryRun: z.boolean(),
  mainBranch: z.string().min(1, 'main branch is required'),
  packagePatterns: z
    .array(z.string().min(1))
    .min(1, 'at least one package pattern is required'),
  prAssignees: z.array(z.string().min(1)),
  skipTests: z.boolean(),
  skipE2E: z.boolean(),
  token: z.string().min(1, 'token is required')
});
