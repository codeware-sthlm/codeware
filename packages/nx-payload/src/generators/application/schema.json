{
  "title": "Generate a Payload NextJS application",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the application.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the application?",
      "pattern": "^[a-zA-Z].*$"
    },
    "directory": {
      "description": "The path to the new application.",
      "type": "string",
      "x-prompt": "Where would you like to create the application files?"
    },
    "database": {
      "description": "Database adapter to setup initially.",
      "type": "string",
      "enum": ["mongodb", "postgres"],
      "default": "mongodb"
    },
    "e2eTestRunner": {
      "type": "string",
      "enum": ["playwright", "none"],
      "description": "Test runner to use for e2e tests.",
      "default": "none"
    },
    "linter": {
      "type": "string",
      "enum": ["eslint", "none"],
      "description": "The tool to use for running lint checks.",
      "default": "eslint"
    },
    "unitTestRunner": {
      "type": "string",
      "enum": ["jest", "none"],
      "description": "Test runner to use for unit tests.",
      "default": "jest"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "skipFormat": {
      "type": "boolean",
      "description": "Skip formatting application files.",
      "default": false
    }
  },
  "required": ["directory", "name"],
  "examplesFile": "../../../docs/application-examples.md"
}
