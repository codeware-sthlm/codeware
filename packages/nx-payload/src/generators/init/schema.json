{
  "title": "Init Nx Payload Plugin",
  "type": "object",
  "properties": {},
  "required": []
}
