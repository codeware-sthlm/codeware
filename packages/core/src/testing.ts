export { createSchemaTests } from './lib/testing/create-schema-tests';
export { SchemaRegistry } from './lib/testing/schema-registry.class';
