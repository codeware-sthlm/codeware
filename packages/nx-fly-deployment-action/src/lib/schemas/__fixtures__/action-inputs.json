{
  "flyApiToken": "fly-token",
  "flyOrg": "company",
  "flyRegion": "sea",
  "mainBranch": "main",
  "env": ["ENV_KEY1=env-value1", "ENV_KEY2=env-value2"],
  "secrets": ["SECRET_KEY1=secret-value1", "SECRET_KEY2=secret-value2"],
  "token": "github-token"
}
