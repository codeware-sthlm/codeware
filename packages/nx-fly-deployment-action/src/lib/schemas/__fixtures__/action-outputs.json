{
  "environment": "production",
  "projects": [
    {
      "action": "deploy",
      "app": "nextgen-web-app",
      "name": "nextgen-web",
      "url": "https://nextgen-web-app.fly.dev"
    },
    {
      "action": "destroy",
      "app": "poc-api-app"
    },
    {
      "action": "skip",
      "appOrProject": "broken-project",
      "reason": "No fly.toml file found"
    },
    {
      "action": "skip",
      "appOrProject": "broken-app",
      "reason": "Failed to deploy application"
    }
  ]
}
