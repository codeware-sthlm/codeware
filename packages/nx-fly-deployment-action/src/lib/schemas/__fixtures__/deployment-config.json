{
  "dryRun": false,
  "fly": {
    "token": "fly-token",
    "org": "company",
    "region": "sea"
  },
  "mainBranch": "main",
  "secrets": {
    "SECRET_KEY1": "secret-value1",
    "SECRET_KEY2": "secret-value2"
  },
  "token": "github-token"
}
