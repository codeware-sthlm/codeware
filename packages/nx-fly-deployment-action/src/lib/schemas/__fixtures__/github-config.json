{
  "deploy": true,
  "flyConfig": "path/to/fly.toml"
}
