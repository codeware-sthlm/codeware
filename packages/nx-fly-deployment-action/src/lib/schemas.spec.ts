import { join } from 'path';

import { SchemaRegistry, createSchemaTests } from '@codeware/core/testing';

import { ActionInputsSchema } from './schemas/action-inputs.schema';
import { ActionOutputsSchema } from './schemas/action-outputs.schema';
import { DeploymentConfigSchema } from './schemas/deployment-config.schema';
import { GitHubConfigSchema } from './schemas/github-config.schema';

// Register all schemas and fixtures for automated testing
SchemaRegistry.register('action-inputs', ActionInputsSchema, {
  name: 'ActionInputsSchema'
});
SchemaRegistry.register('action-outputs', ActionOutputsSchema, {
  name: 'ActionOutputsSchema'
});
SchemaRegistry.register('deployment-config', DeploymentConfigSchema, {
  name: 'DeploymentConfigSchema'
});
SchemaRegistry.register('github-config', GitHubConfigSchema, {
  name: 'GitHubConfigSchema'
});

// Run the tests
createSchemaTests({
  schemaDir: join(__dirname, 'schemas'),
  description: 'Fly Deployment Action schema validation',
  snapshots: true
});
