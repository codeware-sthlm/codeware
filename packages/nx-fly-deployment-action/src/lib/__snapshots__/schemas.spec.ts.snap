// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Fly Deployment Action schema validation > validates all schemas against fixtures > ActionInputsSchema transformed shape 1`] = `
{
  "env": [
    "ENV_KEY1=env-value1",
    "ENV_KEY2=env-value2",
  ],
  "flyApiToken": "fly-token",
  "flyOrg": "company",
  "flyRegion": "sea",
  "mainBranch": "main",
  "secrets": [
    "SECRET_KEY1=secret-value1",
    "SECRET_KEY2=secret-value2",
  ],
  "token": "github-token",
}
`;

exports[`Fly Deployment Action schema validation > validates all schemas against fixtures > ActionOutputsSchema transformed shape 1`] = `
{
  "environment": "production",
  "projects": [
    {
      "action": "deploy",
      "app": "nextgen-web-app",
      "name": "nextgen-web",
      "url": "https://nextgen-web-app.fly.dev",
    },
    {
      "action": "destroy",
      "app": "poc-api-app",
    },
    {
      "action": "skip",
      "appOrProject": "broken-project",
      "reason": "No fly.toml file found",
    },
    {
      "action": "skip",
      "appOrProject": "broken-app",
      "reason": "Failed to deploy application",
    },
  ],
}
`;

exports[`Fly Deployment Action schema validation > validates all schemas against fixtures > DeploymentConfigSchema transformed shape 1`] = `
{
  "fly": {
    "org": "company",
    "region": "sea",
    "token": "fly-token",
  },
  "mainBranch": "main",
  "secrets": {
    "SECRET_KEY1": "secret-value1",
    "SECRET_KEY2": "secret-value2",
  },
  "token": "github-token",
}
`;

exports[`Fly Deployment Action schema validation > validates all schemas against fixtures > GitHubConfigSchema transformed shape 1`] = `
{
  "deploy": true,
  "flyConfig": "path/to/fly.toml",
}
`;
